버튼 클릭 시, 채널 코드를 클립보드에 복사할 수 있도록 하는 CodeShareButton 컴포넌트 구현

always-awake opened the pull request:
# 버튼 클릭 시, 채널 코드를 클립보드에 복사할 수 있도록 하는 CodeShareButton 컴포넌트 구현

## 해당 이슈 📎

- 공유 버튼을 이용해서 생성한 프레젠테이션 채널 코드를 복사하고, 공유할 수 있다. (#51)

## 변경 사항 🛠
- 채널 공유 UI 컴포넌트 구현
<img width="141" alt="스크린샷 2019-12-02 오후 6 15 48" src="https://user-images.githubusercontent.com/40539104/69946872-c6bad280-152f-11ea-8f42-ed939519ef4f.png">

- 채널 공유 버튼을 클릭하면, 접속중인 채널 코드를 복사할 수 있는 기능을 구현하였다.

## 테스트 ✨
> 없음

## 리뷰어 참고 사항 🙋‍♀️
> 없음

always-awake committed:
[ FEATURE ] 버튼 클릭 시, 채널 코드를 클립보드에 복사할 수 있도록 하는 CodeShareButton 컴포넌트 구현 (#51)

Why
- 사용자가 채널 공유 버튼을 클릭하면, 채널 코드를 복사할 수 있어야 한다.

What
- 공통 컴포넌트인 SmallButton 컴포넌트를 이용해 구현하였다.
- handleCodeShareButtonOnclick 함수를 정의하여, 채널 공유 버튼을 클릭했을 때, 채널 코드가 클립보드에 저장될 수 있도록 설정하였다.

always-awake committed:
[ STYLE ] SlideInfo 컴포넌트의 width 길이 수정 및 SlideStatus 컴포넌트 내 Wrapper 스타일 수정

Why
- 넓이가 균일하게 적용되지 않아 정렬이 제대로 되지 않던 것을 수정해야 한다.
- 스타일 컴포넌트 내에 띄어쓰기가 되지 않아 이를 컨벤션에 맞게 수정해야 한다.

What
- SlideInfo 컴포넌트의 width를 100%에서 94%로 수정하였다.
- SlideStatus내 Wrapper 컴포넌트 스타일 내 띄어쓰기를 수정하였다.

devjaewon commented on web/src/components/channel/Slide/SlideInfo/CodeShareButton/CodeShareButton.jsx:
```diff
@@ -0,0 +1,24 @@
+import { useChannelSelector } from '@/hooks';
+
+const CodeShareButton = () => {
+  const channelCode = useChannelSelector((state) => state.channelCode);
```
`Selector` 라는 훅 네이밍이 좋네요! 참고해야겠습니다.

always-awake replied:
감사합니다 👍
(사실 저건 어느 대단한 팀원분이 만들어 놓으신 함수였습니다 🤓)

devjaewon commented on web/src/components/channel/Slide/SlideInfo/CodeShareButton/CodeShareButton.jsx:
```diff
@@ -0,0 +1,24 @@
+const CodeShareButton = () => {
+  const channelCode = useChannelSelector((state) => state.channelCode);
+  const handleCodeShareButtonOnClick = () => {
+    const element = document.createElement('textarea');
```
이 부분은 어떤동작들을 하는건지 직관적으로 알기 어려운데 `copyToClipboard` 와 같은 유틸 함수를 만드는게 좋지 않을까요?

always-awake replied:
넵넵 수정해보겠습니다~👌

always-awake committed:
[ REFACTOR ] handleCodeShareButtonOnClick 함수 내 로직을 따로 분리하여 copyToClipboard 함수로 구현 (#51)

Why
- 프로젝트와 상관없이 사용될 수 있는 로직은 유틸함수로 분리해야 한다.

What
- dom.js 내 param으로 들어온 값을 클립보드에 복사해주는 유틸함수를 구현하였다.
